crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -191,6 +211,76 @@ def clusters_healthy?
       def compute_resources_available?
         Api::Crowbar.compute_resources_available?
       end
+
+      # Check Errors
+      # all of the below errors return an array with the following schema:
+      # [
+      #   {
+      #     code: String,
+      #     data: ... whatever data type ...,
+      #     help: String # "this is how you might fix the error"
+      #   },
+      #   ...
+      # ]
+      def sanity_check_errors
+        return [] if network_checks.empty?
+
+        [
+          {
+            code: :network_checks,
```

Comment:
Sorry for coming too late with this, but when we have a code for each error, wouldn't it make sense to use it as a key for the error submap? Or is there any reason `errors` must be an array?
```
sanity_check_errors
  return {
    :network_checks: {
      :data: network_checks,
      :help: "help text what to do"
    }
  }
end